xformers/components/attention/_sputnik_sparse.py

```diff
@@ -258,7 +258,7 @@ def matmul_with_mask(self, a, b):
         column_indices = self.column_indices
         out = _sddmm.apply(
             a,
-            b.transpose(-2, -1),
+            b.transpose(-2, -1).contiguous(),
```

Comment:
contiguous is only needed here and down, we were forcing this all the time before. Note that with a good projection kernel this could come for free, could be nice

Reply:
maybe you should leave a comment/breadcrumb trail saying that in the code?

Reply:
will do, or maybe just file an issue, basically the projection in the beginning of MHA could be hardened a little for speed